osu.Game/Rulesets/Mods/ModMuted.cs

```diff
@@ -22,27 +27,81 @@ public abstract class ModMuted : Mod
         public override double ScoreMultiplier => 1;
     }
 
-    public abstract class ModMuted<TObject> : ModMuted, IApplicableToDrawableRuleset<TObject>, IApplicableToTrack
+    public abstract class ModMuted<TObject> : ModMuted, IApplicableToDrawableRuleset<TObject>, IApplicableToTrack, IApplicableToScoreProcessor
         where TObject : HitObject
     {
-        private readonly BindableNumber<double> volumeAdjust = new BindableDouble();
+        private readonly BindableNumber<double> mainVolumeAdjust = new BindableDouble(0.5);
+        private readonly BindableNumber<double> metronomeVolumeAdjust = new BindableDouble(0.5);
 
-        [SettingSource("Enable metronome", "Add a metronome to help you keep track of the rhythm.")]
+        private BindableNumber<int> currentCombo;
+
+        private AudioContainer metronomeContainer;
+
+        [SettingSource("Enable metronome", "Add a metronome beat to help you keep track of the rhythm.")]
         public BindableBool EnableMetronome { get; } = new BindableBool
         {
             Default = true,
             Value = true
         };
 
+        [SettingSource("Final volume at combo", "The combo count at which point the track reaches its final volume.")]
+        public BindableInt MuteComboCount { get; } = new BindableInt
+        {
+            Default = 100,
+            Value = 100,
+            MinValue = 0,
+            MaxValue = 500,
+        };
+
+        [SettingSource("Start muted", "Increase volume as combo builds.")]
+        public BindableBool InverseMuting { get; } = new BindableBool
```

Comment:
This makes sense as being a different mod ("unmute" or "amplify" or otherwise) stacked with muted mod instead, with the implementation being shared between the two. 

That's for consistency with other mod stacks like "Wind Up & Wind Down" and "Grow & Deflate".

Reply:
It's intentionally put in this mod.